src/lib/core/utils/data-element-constraints.ts

```diff
@@ -62,6 +63,7 @@ export function excludedVariables(
  */
 function variableConstraintPredicate(
   constraint: DataElementConstraint,
+  //maybe this should be variable: Variable - from dave
```

Comment:
We don't need to do this, since we are testing against variables that have `type = 'category'`. See comment above.

Reply:
is there any reason to prefer one method over the other? I would have guessed, possibly naively, that just explicitly handling only Variables rather than VariableTreeNodes would be preferred to testing for the condition of meeting a category. what if we introduce some new thing, other than categories, than can be members of VariableTreeNode and not Variable?

Reply:
They are really equivalent. In both cases, we would have to deal with categories. The way we have it now simplifies the calling code, and gives us an opportunity to apply constraints to categories, if we decide to in the future.